### PR TITLE
core(doctype): check document.compatMode for quirks mode

### PR DESCRIPTION
**Pull request by k99sharma**
**Summary**
This change adds a new check in `doctype` audit for `compatMode` of the document. It is needed to check for `quirks-mode` of the document.
I read about quirks-mode check in this MDN doc: [doc](https://developer.mozilla.org/en-US/docs/Web/API/Document/compatMode)

**Related Issues/PRs**
Issue no: #10030

### CODE REVIEW

**Comment by k99sharma** [comment]
I changed the `doctype` audit to check for document quirks-mode. I also changed `getDoctype` function in `gatherer` file to return `compatMode`.

File: types/artifacts.d.ts [hunk 1]
```diff
@@ -17,11 +17,11 @@ type _TaskNode = import('../lighthouse-core/lib/tracehouse/main-thread-tasks.js'
 import AuditDetails from './lhr/audit-details'
 import Config from './config';
 import Gatherer from './gatherer';
-import {IcuMessage} from './lhr/i18n';
+import { IcuMessage } from './lhr/i18n';
```
Comment (patrickhulce) [inline types/artifacts.d.ts:20]: would you revert all of these formatting changes please? makes it hard to tell what the underlying change is :)

Comment (k99sharma) [reply types/artifacts.d.ts:20]: My IDE automatically converted these I will change them back!

Comment (k99sharma) [reply types/artifacts.d.ts]: I was not able to revert it. But I deleted that file and push another commit with the required change only.

**Comment by connorjclark** [comment]
Hey @k99sharma would you be able to fix the merge conflicts?

**Comment by paulirish** [comment]
> Hey @k99sharma would you be able to fix the merge conflicts?

i took care of 'em. and fixed up a few other pieces.

File: lighthouse-core/audits/dobetterweb/doctype.js [hunk 2]
```diff
@@ -76,14 +79,23 @@ class Doctype extends Audit {
...
-        score: 1,
+        score: 0,
+        explanation: str_(UIStrings.explanationBadDoctype),
       };
-    } else {
+    }
+
+    // Catch-all for any quirks-mode situations the above checks didn't get.
+    // https://github.com/GoogleChrome/lighthouse/issues/10030
+    if (compatMode === 'BackCompat') {
```
Comment (connorjclark) [inline lighthouse-core/audits/dobetterweb/doctype.js:91]: I'm thinking we should just _only_ check this.

Comment (connorjclark) [reply lighthouse-core/audits/dobetterweb/doctype.js:91]: Actually... I guess the specific messages are better for diagnosing. nevermind.

**Review by connorjclark — APPROVED** [review]

**Pull request merged**